ruby_2.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    i = 0
```

Comment:
Let's add `return to_enum unless block_given?` here.

```suggestion
    return to_enum unless block_given?
    i = 0
```

Let's do this for other related methods. `my_select`  and  `my_each_with_index`.
Check here https://ruby-doc.org/core-2.6.4/Enumerable.html for more details.